akka-stream-tests/src/test/scala/akka/stream/io/FileSourceSpec.scala

```diff
@@ -59,6 +59,20 @@ class FileSourceSpec extends StreamSpec(UnboundedMailboxConfig) {
     f
   }
 
+  val directoryInsteadOfFile = {
+    val f = Files.createTempDirectory(fs.getPath("/"), "directory-instead-of-file")
+    f
+  }
+
+  val testFifo = {
+    val d = Files.createTempDirectory(fs.getPath("/"), "file-source-spec")
+    val f = d.resolve("test-fifo")
+    // Java or NIO does not supply a way to create named pipes/FIFOs, so we use the POSIX command `mkfifo`.
+    Runtime.getRuntime.exec(s"mkfifo $f")
```

Comment:
Does this actually work? `fs` is an in-memory filesystem created with `JimFs`; it looks like this is just going to create a fifo on the actual filesystem (and it probably won't have permission, nor be portable)

Reply:
Oops, I did not realise `fs` was special - and that is probably the nail in the coffin for this test. [I was already concerned with the portability of this test on at least Windows](https://github.com/akka/akka/pull/25333#issuecomment-403783216) - so I'm wondering if we should even have a test for FIFOs specifically?

Reply:
Yeah I was about to comment on portability but then I checked the source file to see what `fs` was (because I noticed it using `/` and it seemed very unlikely to me that the akka test suite would try to create things in the root filesystem) and discovered JimFs.

I had a look at JimFs and didn't see any mention of fifos so it doesn't look very straightforward to test in a portable way. It'd be a shame to not be able to test this, though, as it means there's no guarantee your fifo will continue working so your issue may recur.

Reply:
Unfortunately it doesn't look like `FileSource` is structured in a way which would make injecting a mock `File` into it very easy either, though, and such a test would stand out as being quite different to the others.

Maybe someone on the akka team has a better suggestion.